### PR TITLE
EDU-995 Extend post-auth logic for external accounts

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-995

### CODE REVIEW

File: src/services/user-service.spec.js [hunk 1]
```diff
@@ -52,6 +52,78 @@ describe('user-service', () => {
     sandbox.restore();
   });
 
+  describe('findConfirmedActiveUserById', () => {
```
Comment (adrianaluchian) [inline src/services/user-service.spec.js:55]: We could still throw the `expires[On]` case in here, in both methods 😅

Comment (ahelmberger) [reply src/services/user-service.spec.js:55]: I'm not sure what you mean :thinking:

Comment (adrianaluchian) [reply src/services/user-service.spec.js:55]: The method has this part that created the bug fixed today:
```
   if (!user || user.expiresOn) {
      return null;
    }
```
with `expires` vs `expiresOn` (sorry I made it look like an array item accessor with `expires[On]`, I meant +/- the `On` part)

Comment (ahelmberger) [reply src/services/user-service.spec.js:55]: I'm not sure if we're already taking about the same thing. But it's a good idea to add test cases for both `findConfirmedActiveUserById` and `findConfirmedActiveUserByEmailAndPassword` that check that no user with `expiresOn` is let through by the query. I'll do that :+1:

Comment (ahelmberger) [reply src/services/user-service.spec.js:55]: I have added those test cases.

Comment (adrianaluchian) [reply src/services/user-service.spec.js:55]: Perfect! 👌

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**